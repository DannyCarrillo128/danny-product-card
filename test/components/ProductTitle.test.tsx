import React from 'react';
import renderer from 'react-test-renderer';

import { ProductCard, ProductTitle } from '../../src/components';
import { product1 } from '../data/products';

describe('Product Title', () => {

  test('should show the component with the corresponding title', () => {
    const wrapper = renderer.create(
      <ProductTitle className="custom-class" title="sample text" />
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('should show the component with the product name', () => {
    const wrapper = renderer.create(
      <ProductCard product={ product1 }>
        {
          () => (
            <ProductTitle />
          )
        }
      </ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  
});
