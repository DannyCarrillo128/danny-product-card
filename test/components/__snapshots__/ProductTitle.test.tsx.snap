// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Title should show the component with the corresponding title 1`] = `
<span
  className="productDescription custom-class"
>
  sample text
</span>
`;

exports[`Product Title should show the component with the product name 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee mug
  </span>
</div>
`;
